### ISSUE TITLE
OS X 10.6 buildbot is failing with 'interrupted system call' errors

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6247]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)| @glyph reported|
|-|-|
|Trac ID|trac#6247|
|Type|enhancement|
|Created|2013-01-09 23:44:03Z|
|Branch|https://github.com/twisted/twisted/tree/release-test-failures-6247|

We thought this was due to failing hardware, but the "failing hardware" was actually a pypy/buildbot incompatibility.  It should be fixed.

<details><summary>Searchable metadata</summary>

```
trac-id__6247 6247
type__enhancement enhancement
reporter__glyph glyph
priority__normal normal
milestone__ 
branch__branches_release_test_failures_6247 branches/release-test-failures-6247
branch_author__glyph glyph
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1357775043000000 1357775043000000
changetime__1358061724000000 1358061724000000
version__None None
owner__glyph glyph

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_1"></a>|@glyph commented|
|-|-|

(In [36766]) Branching to 'release-test-failures-6247'

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_2"></a>|@glyph removed owner|
|-|-|

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/release-test-failures-6247) for the branch indicate that it fixes the issue, and doesn't cause any related problems.  (I checked the FreeBSD builder and it is now failing on trunk as well, not due to a code change, but due to a build configuration issue; already contacted the maintainer.)

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_3"></a>|@tomprince set owner to @glyph|
|-|-|

Looks good. Please commit.

Cursory research indicates that the origin of the code might be due to http://bugs.python.org/issue1068268 which is fixed in 2.6 (being the oldest supported version now).

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_4"></a>|@glyph set status to `closed`|
|-|-|

(In [36804]) Merge release-test-failures-6247.

Author: glyph

Reviewer: tom.prince

Fixes: [#6247](https://github.com/twisted/twisted/issues/6247)

Remove signal handler manipulation from tests, correcting test failures on OS X
10.6.
